stl/inc/expected

```diff
@@ -1120,18 +1112,18 @@ private:
     // These overloads force copy elision from the invoke call into _Value
     template <class _Fn, class _Ux>
     constexpr expected(_Construct_expected_from_invoke_result_tag, _Fn&& _Func, _Ux&& _Arg) noexcept(
-        noexcept(_Ty(_STD invoke(_STD forward<_Fn>(_Func), _STD forward<_Ux>(_Arg)))))
+        noexcept(static_cast<_Ty>(_STD invoke(_STD forward<_Fn>(_Func), _STD forward<_Ux>(_Arg)))))
         : _Value(_STD invoke(_STD forward<_Fn>(_Func), _STD forward<_Ux>(_Arg))), _Has_value{true} {}
 
-    // For when transform is called on an expected<void,E>
+    // For when transform is called on an expected<void, E> and requires calling _Func with no arg
     template <class _Fn>
     constexpr expected(_Construct_expected_from_invoke_result_tag, _Fn&& _Func) noexcept(
-        noexcept(_Ty(_STD forward<_Fn>(_Func)())))
-        : _Value(_STD forward<_Fn>(_Func)()), _Has_value{true} {}
+        noexcept(static_cast<_Ty>(_STD forward<_Fn>(_Func)()))) // equivalent to invoke()

```

Comment:
This is most definitely _not_ "equivalent to `invoke()`" `invoke()` is ill-formed. (Several occurrences) I'll push a fix.